### ISSUE TITLE
[Bug] ROI tools: Can't chose the ROI shape before clicking add ROI, and sometimes wrong one is shown after selection. 

### ISSUE THREAD

**Issue by mathieuboudreau** [qMRLab/qMRLab#232]
It only gets enabled after I click "Add ROI", but if I then change the shape option before drawing the ROI, I sometimes can't draw it and then get a lot of errors in the Matlab window. I've tried it a few times, I can send a video capture if you want. I get these types of errors after I test around for a bit:

```
Error using iptSetPointerBehavior>parseInputs (line 134)
First input argument, h, contains one or more invalid handles.

Error in iptSetPointerBehavior (line 118)
[h, pointerBehavior] = parseInputs(varargin{:});

Error in manageInteractivePlacement/resetChildPointerManagement (line 154)
              iptSetPointerBehavior(cached_pointer_behavior{k,1},...

Error in manageInteractivePlacement (line 105)
      resetChildPointerManagement();

Error in imrect>imrectAPI (line 275)
    placement_aborted =
    manageInteractivePlacement(h_axes,h_group,@placeRect);

Error in imrect (line 82)
            [h_group,draw_api] = imrectAPI(varargin{:});

Error in Roi_analysis>drawROI_Callback (line 149)
        draw = imrect();

Error in Roi_analysis>AddROI_Callback (line 119)
drawROI_Callback(hObject, eventdata, handles)

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in Roi_analysis (line 30)
    gui_mainfcn(gui_State, varargin{:});

Error in
matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)Roi_analysis('AddROI_Callback',hObject,eventdata,guidata(hObject)) 
Error while evaluating UIControl Callback.

```

**Comment by mathieuboudreau** [comment]
Oooh ✨, great find @tanguyduval ! We should check down to which MATLAB version this tool is compatible (I see it was designed for 2016b), but this certainly looks like a great tool to include into qMRLab!

**Comment by TommyBoshkovski** [comment]
Great find indeed @tanguyduval ! I will check for dependencies and the compatibility of the tool and I'll include into qMRLab.

**Comment by tanguyduval** [comment]
@TommyBoshkovski I just run a dependency report and looks good now.
We have 2 files already on qMRLab: find3d.m and range_outlier.m. we might have to remove them to prevent conflicts.
Best

**Comment by tanguyduval** [comment]
@mathieuboudreau I'm using R2018.... cannot guarantee it's working on earlier versions

**Comment by mathieuboudreau** [comment]
@tanguyduval if we're using those functions, no need to remove them - just rename the files/functions and wherever they're called. (Isn't range_outlier the function that you yourself just recently wrote for anogher PR?)

**Comment by tanguyduval** [comment]
yes, I wrote both of them

**Comment by agahkarakuzu** [comment]
@tanguyduval  blessing to your search and your code :) 

Throwing this just as an idea, maybe we can play with at hackathon. 3D Slicer has a [Matlab bridge](https://www.slicer.org/wiki/Documentation/4.3/Extensions/MatlabBridge), that supposed to "seamlessly" execute a matlab module (if matlab exec is available of course). If it works, we can suggest this as an alternative, given that 3D Slicer is equipped with all fancy ITK/VTK abilities.

**Comment by tanguyduval** [comment]
@agahkarakuzu thanks for sharing this functionality of 3D Slicer 👍 
That would mean putting qMRLab inside 3D Slicer, not the inverse right? I guess this is more an additional project than an alternative though (if we want qMRLab to work independently).

**Comment by agahkarakuzu** [comment]
@tanguyduval as you said, just a side project to play with. If it works maybe we can create some tutorials with it, some people may benefit from it (maybe, as usual :) ).

**Comment by tanguyduval** [comment]
@TommyBoshkovski have you started to implement imtool3d to qMRlab? I can help for the first steps.. I should do a brief intro to you by skype someday

**Comment by TommyBoshkovski** [comment]
@tanguyduval I started integrating the imtool to qMRLab, but then I started working on the website and I haven't finished the integration. I'll finish it ASAP. However, If I run into a prblem then we can do a Skype meeting.

**Comment by tanguyduval** [comment]
Good! Just wanted to make sure that you use the functionality that it can be added directly in the panel at creation:
```    
%tool = imtool3D(I,position,h) puts the imtool3D panel in the figure
 %specified by the handle h.
```
And then you interact with it using the methods... No need to copy code or even code anything ;)

**Comment by tanguyduval** [comment]
@TommyBoshkovski I'm starting a branch with imtool3D.  I will replace the current viewer, so it's mostly cleaning and pluging the viewer with qMRLab data. A few hours maybe.... 

What's your status before I start?? I just saw that you replaced the ROI Analysis with imtool3D. Do you have any addidional modifications??
Best

**Comment by TommyBoshkovski** [comment]
@tanguyduval I didn't do any modifications on the imtool3D. I just integrated it in the qMRLab, so now qMRLab is using imtool3D. 
Best

**Comment by tanguyduval** [comment]
solved in #296
